EmployeeList/Utils/APIRequest.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  APIService.swift
+//  EmployeeList
+//
+//  Created by Wang Siman on 1/29/21.
+//  Copyright © 2021 None. All rights reserved.
+//
+
+import Foundation
+
+protocol APIRequestType {
```

Comment:
I would probably just add an extension to URLRequest to load it with these values; the contents are pretty much the same, no?

Reply:
I read some blogs about protocol-oriented networking and think it woud be a good practice to define a general request protocol, such that it allows me to create as many types of requests as I need. `ModelType` specifies the type of data for which the request asks. With this associatedtype, I'm able to define a generic funciont `call()` in APIServiceType protocol to return a `AnyPublisher` carrying decoded data of ModelType. 

`func call<Request>(from endpoint: Request) -> AnyPublisher<Request.ModelType, Error> where Request: APIRequestType`
This benefits me in making mockAPIService in unit testing. 

Actually, at the beginning, I didn't use the associatedtype ModelType in APIRequestType, and call() was like
`func call<Value>(from endpoint: Request) -> AnyPublisher<Value, Error> where Value: Decodable`
I found it was hard, in unit testing, to re-implement call() in mockAPIService that returns a mock response, which was hard-coded with a specific data type (e.g. EmployeeListResponse) in test cases.

`queryItems` allows me to specify queries easily. 

I think a cleaner way would be:
```
protocol APIRequestType {
    associatedtype ModelType: Decodable
    
    var urlRequest: URLRequest { get }
    var queryItems: [URLQueryItem]? { get }
}
```

I don't know how adding an extension to URLRequest can do all these work. Please show me the best practice!!!